libs/gi/db/src/Database/DataManagers/ArtifactDataManager.ts

```diff
@@ -215,6 +215,8 @@ export class ArtifactDataManager extends DataManager<
           importArt = {
             ...art,
             location: hasEquipment ? art.location : match.location,
+            // Preserve existing lock state for artifacts reported as unlocked or when the scanner doesn't detect locks ("lock": false JSON)
+            lock: art.lock === false ? match.lock : art.lock,
```

Comment:
I would preferred if we do 
```
const hasLock = artifacts.some((a)=>a.lock)
...
lock: hasLock? art.lock : match.lock
```
similiar to `hasEquipment`

since this logic would not allow unlocking of artifacts from the scanned data (if user unlocked in game, and scanner scanned it, but it's locked in GO)

Reply:
I made the requested change and confirmed with testing where having all the lock properties as false keeps the original lock statuses (locked/unlocked), and if there are locks present, artifacts can become be locked or unlocked